cabal.project

```diff
@@ -171,8 +174,7 @@ source-repository-package
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-node
-  tag: 1.35.1
-  --sha256: 1z0zv1i58ikmbqg878f9z573jkwp4lzhmmswshm6c96rq6lprzh8
+  tag: 781ff37e5acb0c59034f67f57c9e639161be9d96
```

Comment:
You can update to tag `1.35.2`.

Reply:
I can't, because the other deps are outdated. 

cardano-node's `1.35.2` uses `cardano-ledger`'s commit `ebcf1a8936dd84de0182d54004473f4ce66c7923` that  fails with

```
src/Cardano/Crypto/Hashing.hs:122:15: error:
    • Could not deduce (B.Buildable b0)
        arising from a use of ‘toAesonError’
      from the context: HashAlgorithm algo
        bound by the instance declaration
        at src/Cardano/Crypto/Hashing.hs:121:10-61
      The type variable ‘b0’ is ambiguous
      These potential instances exist:
        instance B.Buildable Void -- Defined in ‘Formatting.Buildable’
        instance B.Buildable IntPtr -- Defined in ‘Formatting.Buildable’
        instance B.Buildable WordPtr -- Defined in ‘Formatting.Buildable’
        ...plus 25 others
        ...plus 19 instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the first argument of ‘(.)’, namely ‘toAesonError’
      In the first argument of ‘(<=<)’, namely
        ‘toAesonError . readEither’
      In the expression: toAesonError . readEither <=< parseJSON
    |
122 |   parseJSON = toAesonError . readEither <=< parseJSON
    |               ^^^^^^^^^^^^

src/Cardano/Crypto/Hashing.hs:122:30: error:
    • Could not deduce (protolude-0.3.0:Protolude.Conv.StringConv
                          b0 Prelude.String)
        arising from a use of ‘readEither’
      from the context: HashAlgorithm algo
        bound by the instance declaration
        at src/Cardano/Crypto/Hashing.hs:121:10-61
      The type variable ‘b0’ is ambiguous
      These potential instances exist:
        instance protolude-0.3.0:Protolude.Conv.StringConv
                   ByteString Prelude.String
          -- Defined in ‘protolude-0.3.0:Protolude.Conv’
        instance protolude-0.3.0:Protolude.Conv.StringConv
                   LBS.ByteString Prelude.String
          -- Defined in ‘protolude-0.3.0:Protolude.Conv’
        instance protolude-0.3.0:Protolude.Conv.StringConv
                   Text Prelude.String
          -- Defined in ‘protolude-0.3.0:Protolude.Conv’
        ...plus one other
        ...plus one instance involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the second argument of ‘(.)’, namely ‘readEither’
      In the first argument of ‘(<=<)’, namely
        ‘toAesonError . readEither’
      In the expression: toAesonError . readEither <=< parseJSON
    |
122 |   parseJSON = toAesonError . readEither <=< parseJSON
    |                              ^^^^^^^^^^

src/Cardano/Crypto/Hashing.hs:122:45: error:
    • Could not deduce (FromJSON b0) arising from a use of ‘parseJSON’
      from the context: HashAlgorithm algo
        bound by the instance declaration
        at src/Cardano/Crypto/Hashing.hs:121:10-61
      The type variable ‘b0’ is ambiguous
      These potential instances exist:
        instance (FromJSON a, FromJSON b) => FromJSON (Either a b)
          -- Defined in ‘aeson-2.0.2.0:Data.Aeson.Types.FromJSON’
        instance forall k a (b :: k). FromJSON a => FromJSON (Const a b)
          -- Defined in ‘aeson-2.0.2.0:Data.Aeson.Types.FromJSON’
        instance FromJSON a => FromJSON (Identity a)
          -- Defined in ‘aeson-2.0.2.0:Data.Aeson.Types.FromJSON’
        ...plus 50 others
        ...plus 56 instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the second argument of ‘(<=<)’, namely ‘parseJSON’
      In the expression: toAesonError . readEither <=< parseJSON
      In an equation for ‘parseJSON’:
          parseJSON = toAesonError . readEither <=< parseJSON
    |
122 |   parseJSON = toAesonError . readEither <=< parseJSON
    |                                             ^^^^^^^^^
```

because we use too recent aeson I suppose. They have patched it but in a more recent version (https://github.com/input-output-hk/cardano-ledger/commit/b425b51a80f57cadccff5eaf9443bf492b2c51d6#diff-80065f789ee866e5fe987091d83f0ee7354182be0161d8cd3dc087f0c48a2a36R123).

By updating the `cardano-ledger`, I also need to update the `ouroboros-network` and `cardano-node` as well. They are linked together.

Reply:
Might to "too recent aeson" be because the `index-state` was updated? Shouldn't we use the same index-state as our dependencies to prevent this?